strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/AbstractRepositoryProvider.java

```diff
@@ -140,14 +149,15 @@ protected RepositoryInputStream decorate(String storageId,
 
     @Override
     public RepositoryOutputStream getOutputStream(Path path)
-        throws IOException,
-        NoSuchAlgorithmException
+        throws IOException
     {
         Assert.isInstanceOf(RepositoryPath.class, path);
-        return getOutputStream((RepositoryPath) path);
+        OutputStream os = getOutputStreamInternal((RepositoryPath) path);
+        
+        return decorate((RepositoryPath) path, os);
     }
     
-    protected abstract RepositoryOutputStream getOutputStream(RepositoryPath repositoryPath)
+    protected abstract OutputStream getOutputStreamInternal(RepositoryPath repositoryPath)
```

Comment:
Why is it necessary to rename these methods?

Reply:
because there is two `getOutputStream` methods and it causes a confusions and ambiguity, some tests was failed because wrong `getOutputStream` method was invoked within test artifacts generation.

Reply:
I see. Thanks for explaining! :)